File: driver_app/lib/domain/entities/delivery.dart
import 'package:equatable/equatable.dart';

class Delivery extends Equatable {
  final String id;
  final String orderId;
  final String locationId;
  final String status;
  final String recipientName;
  final String recipientPhone;
  final String recipientAddress;
  final double deliveryFee;
  final List<DeliveryEvent> timeline;
  final Map<String, dynamic>? proof;

  const Delivery({
    required this.id,
    required this.orderId,
    required this.locationId,
    required this.status,
    required this.recipientName,
    required this.recipientPhone,
    required this.recipientAddress,
    required this.deliveryFee,
    required this.timeline,
    this.proof,
  });

  @override
  List<Object?> get props => [id, status, timeline, proof];
}

class DeliveryEvent extends Equatable {
  final String status;
  final DateTime timestamp;
  final String? note;

  const DeliveryEvent({
    required this.status,
    required this.timestamp,
    this.note,
  });

  @override
  List<Object?> get props => [status, timestamp, note];
}

File: functions/src/models/delivery.ts
import { Timestamp } from 'firebase-admin/firestore';

export type DeliveryStatus =
    | 'queued'
    | 'assigned'
    | 'picked_up'
    | 'in_transit'
    | 'delivered'
    | 'failed'
    | 'returned';

export interface Delivery {
    id: string;
    orderId: string;
    city: string;
    areaId: string;
    areaName: string;
    locationId?: string;
    status: DeliveryStatus;

    assignedDriver?: {
        uid: string;
        name: string;
        phone: string;             // Never exposed to customers
    };

    internalNotes?: string;        // Admin/ops only — never exposed to customers

    pickup: {
        storeId: string;
        address: string;
        scheduledAt: Timestamp;
        actualAt?: Timestamp;
    };

    dropoff: {
        address: string;
        recipientName: string;
        recipientPhone: string;   // Never exposed to customers
        actualAt?: Timestamp;
    };

    timeline: DeliveryTimelineEvent[];

    proof?: {
        /** Storage path — NOT a public URL. Access via signed URL endpoint. */
        storagePath?: string;
        otpVerified?: boolean;
        signedAt?: Timestamp;
    };

    deliveredAt?: Timestamp;

    createdAt: Timestamp;
    updatedAt: Timestamp;
}

export interface DeliveryTimelineEvent {
    status: DeliveryStatus;
    timestamp: Timestamp;
    note?: string;
    location?: { lat: number; lng: number };
}

// ─── Customer-safe public view ────────────────────────────────────────────────
/** Stored in delivery_public/{orderId}. Safe for customers to read directly. */
export interface DeliveryPublicView {
    orderId: string;
    status: DeliveryStatus;
    timeline: Array<{ status: DeliveryStatus; timestamp: Timestamp }>;
    proofSummary: {
        otpVerified: boolean;
        photoUploaded: boolean;
    };
    deliveredAt?: Timestamp;
    updatedAt: Timestamp;
}
